operatorcourier/validate.py

```diff
@@ -237,3 +245,135 @@ def _type_validation(self, bundleData, typeName, validator, required):
+                valid = False
+            
+            if valid:
+                if self._ui_csv_fields_format_validation(csv) is False:
+                    logger.error("UI validation failed to verify required fields exist.")
+                    valid = False
+
+        return valid
+
+    def _ui_csv_fields_exist_validation(self, csv):
+        valid = True
+
+        if "metadata" not in csv:
+            logger.error("csv metadata not defined.")
+            valid = False
+        else:
+            if "name" not in csv["metadata"]:
+                logger.error("csv metadata.name not defined.")
+                valid = False
+            else:
+                logger.info("Evaluating csv %s", csv["metadata"]["name"])
+
+                for field in general_required_fields:
+                    if field not in csv:
+                        logger.error("csv %s not defined.", field)
+                        valid = False
+                        return valid
+                
+                for field in metadata_required_fields:
+                    if field["field"] not in csv["metadata"]:
+                        if field["required"]:
+                            logger.error("csv metadata.%s not defined. %s", field["field"], field["description"])
+                            valid = False
+                            return valid
+                        else:
+                            logger.warning("csv metadata.%s not defined. %s", field["field"], field["description"])
+                
+                for field in metadata_annotations_required_fields:
+                    if field["field"] not in csv["metadata"]["annotations"]:
+                        if field["required"]:
+                            logger.error("csv metadata.annotations.%s not defined. %s", field["field"], field["description"])
+                            valid = False
+                        else:
+                            logger.warning("csv metadata.annotations.%s not defined. %s", field["field"], field["description"])
+
+                for field in spec_required_fields:
+                    if field["field"] not in csv["spec"]:
+                        if field["required"]:
+                            logger.error("csv spec.%s not defined. %s", field["field"], field["description"])
+                            valid = False
+                        else:
+                            logger.warning("csv spec.%s not defined. %s", field["field"], field["description"])
+
+        return valid
+    
+    def _ui_csv_fields_format_validation(self, csv):
+
+        def is_url(field):
+            if not v.url(field):
+                return False
+            return True
+
+        def is_email(field):
+            if not v.email(field):
+                return False
+            return True
+
+        def is_version(field):
+            pattern1 = re.compile(r'v(\d+\.)(\d+\.)(\d)')
```

Comment:
I believe version also supports things like `-alpha` that can be added to the end of a semver.
Can these patterns be consolidated? Eg: `r'v?(\d+\.)(\d+\.)(\d)(-[a-z]+)?'`

Reply:
Good point. @jeff-phillips-18 can you confirm that this is allowed for the UI?

Reply:
Actually it may be better to check with this package: https://pypi.org/project/semver/
We should use a regex that matches what quay.io is using to be complete (the above isn't that :/).

Reply:
Yes that would work for the UI. We only really use it to sort the different versions, for version selection dropdown and for determining the latest version.